### ISSUE TITLE
Interest in merging with scikit-sparse?

### ISSUE THREAD

**Issue by anntzer** [scikit-umfpack/scikit-umfpack#26]
I recently inherited the maintenance ("intended as bugfix-only") of `scikit-sparse` (https://github.com/scikit-sparse/scikit-sparse, http://pythonhosted.org/scikit-sparse/), a wrapper for the `cholmod` part of suitesparse (Cholesky decomposition) initially written by njsmith.  It's not a very big package, and given that both scikit-sparse and scikit-umfpack exist primarily (AFAICT) because of suitesparse's license, perhaps it may make sense to merge them?

See also scikit-sparse/scikit-sparse#11.

**Comment by rc** [comment]
+1, it makes sense to have single wrappers of suitesparse code. But note that the low level scikit-umfpack code is very old and uses swig to interface with umfpack C libraries instead of Cython, so it would take some work and time to modernize that. The code was originally a part of scipy.sparse, and the scikit was created relatively not so long ago because of suitesparse's license as you say. I have resources to do a basic maintenance work, but cannot (at the moment) devote time to the merge.

As for https://github.com/scikit-sparse/scikit-sparse/issues/11, note that @pv later created a very nice high-level interface for scikit-umfpack.

What we could do now is to plan here the steps necessary for the possible future merge. The first logical step would be to unite the wrapping machinery. What do you think?

**Comment by anntzer** [comment]
Can you provide a pointer to @pv's interface?  Perhaps it may make more sense to include his code in scikit-sparse instead, depending on how much it provides compared to scikit.umfpack.

**Comment by pv** [comment]
It's a part of scikits.umfpack in the recent releases.

**Comment by anntzer** [comment]
OK.
Unfortunately, it seems that there is very little to be shared between choldmod and umfpack (for example, only cholmod uses a sparse matrix struct whereas umfpack passes multiple arrays around), so "uniting the wrapping machinery" may be more a grandiose plan than anything else (or do you have specific thoughts?).
It may make more sense to restart from the [umfpack branch of scikit-sparse](https://github.com/scikit-sparse/scikit-sparse/tree/umfpack) make sure that @pv's API is available there too.
Anyways, I'll just let this sit for now, if someone want to pick it up...

**Comment by rc** [comment]
Yes, it would be a significant work. Maybe a good topic for next year's gsoc :)

**Comment by AndyJenks** [comment]
It looks like this is going to happen at the scipy end, as of the last year or so: https://github.com/scipy/scipy/pull/24101
